plugins/wasi_nn/ggml.cpp

```diff
@@ -483,29 +494,25 @@ Expect<ErrNo> getEmbedding(WasiNNEnvironment &Env,
 
   // Get the context size.
   const uint64_t NCtx = llama_n_ctx(LlamaContext);
-  // Minus 4 for the special tokens. (Such as <BOS>, <EOS>, ... tokens.)
-  const uint64_t MaxTokensListSize = NCtx - 4;
   // Use the const sequence id here.
   const llama_seq_id SequenceId = 0;
   // Return value.
   auto ReturnCode = ErrNo::Success;
 
-  // Add BOS if not present.
-  if (CxtRef.LlamaInputs.front() != llama_token_bos(GraphRef.LlamaModel)) {
-    CxtRef.LlamaInputs.insert(CxtRef.LlamaInputs.begin(),
-                              llama_token_bos(GraphRef.LlamaModel));
-  }
-  // Add EOS if not present.
-  if (!llama_token_is_eog(GraphRef.LlamaModel, CxtRef.LlamaInputs.back())) {
-    CxtRef.LlamaInputs.push_back(llama_token_eos(GraphRef.LlamaModel));
+  // Add SEP if not present.
+  if (CxtRef.LlamaInputs.back() != llama_token_sep(GraphRef.LlamaModel)) {
+    CxtRef.LlamaInputs.push_back(llama_token_sep(GraphRef.LlamaModel));
   }
 
   // Check if the input is too long.
-  if (static_cast<uint64_t>(CxtRef.LlamaInputs.size()) > MaxTokensListSize) {
+  if (static_cast<uint64_t>(CxtRef.LlamaInputs.size()) >
+      ContextParams.n_batch) {
```

Comment:
Do we need to evaluate the `n_ubatch` here?

Reply:
I think we could just follow this to set `n_ubatch = n_batch` when generating embeddings. I'll add this.
https://github.com/ggerganov/llama.cpp/blob/1b67731e184e27a465b8c5476061294a4af668ea/examples/embedding/embedding.cpp#L64-L65